actix-http/src/extensions.rs

```diff
@@ -61,6 +61,11 @@ impl Extensions {
     pub fn clear(&mut self) {
         self.map.clear();
     }
+
+    /// Extends it with the content for another `Extensions`
+    pub fn extend(&mut self, extensions: Extensions) {
```

Comment:
Wouldn't implementing [`std::iter::Extend`](https://doc.rust-lang.org/std/iter/trait.Extend.html) make more sense?

Also, unit-tests for this functionality would be appreciated.

Reply:
I agree. Though I'm not being able to make it compile... As I wanted to extend with another object of the same type and not with the internal structure...

```rust
impl Extend<Extensions> for Extensions {

    fn extend<T: IntoIterator<Item = Extensions>>(&mut self, iter: T) {
        self.map.extend(iter);
    }
}

error[E0271]: type mismatch resolving `<T as std::iter::IntoIterator>::Item == (std::any::TypeId, std::boxed::Box<(dyn std::any::Any + 'static)>)`
  --> actix-http/src/extensions.rs:75:18
   |
75 |         self.map.extend(iter);
   |                  ^^^^^^ expected struct `extensions::Extensions`, found tuple
   |
   = note: expected struct `extensions::Extensions`
               found tuple `(std::any::TypeId, std::boxed::Box<(dyn std::any::Any + 'static)>)`
```

Reply:
I believe it would require you to implement `IntoIterator` for `Extensions` as well.

Reply:
@robjtede what is your opinion here? should we implement `IntoIterator` for `Extensions`?

Reply:
the current seems fine to me, it's not really useful on the developer side

Reply:
I'm not insisting either.